### ISSUE TITLE
"Unable to open file -- string too long"

### ISSUE THREAD

**Issue by Saguaroman** [jasp-stats/jasp-issues#393]
Hi:  
I'm trying to open an SPSS file in JASP. The data file is attached.  I get the error:    "Unable to open file.  String too long".  I've deleted all the string variables from the file, deleted empty columns and a couple of variables with weird names or long variable labels, but still get the problem.  Thanks for any help you can provide. 

* JASP version: 0.10
* OS name and version: Windows 7 Enterprise - 64 bit
* Analysis: Trying to load file
* Bug description:    "Unable to open file.  String too long" 
* Add JASP files as a zip:  
[FRIENDS Combined SME.zip](https://github.com/jasp-stats/jasp-issues/files/3310852/FRIENDS.Combined.SME.zip)
* Steps to reproduce:  1. Open JASP, 2. try to load data file.

**Comment by JorisGoosen** [comment]
Could you try to export your file to csv from spss?

**Comment by JorisGoosen** [comment]
My colleague tried to load your file in R itself and got quite a load of errors there as well. (Using ` foreign::read.spss(file="~/downloads/FRIENDS Combined SME.sav”)`)

[friendsCom.csv.zip](https://github.com/jasp-stats/jasp-issues/files/3316962/friendsCom.csv.zip)

Maybe you could work from that?
And this file does work without problems in spss?

At a minimum the following errors were found:
```

Warning messages:
1: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SlfDscls1: ., .
2: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SlfDscls2: ., .
3: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SlfDscls3: ., .
4: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor PtrDscls1: ., .
5: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor PtrDscls2: ., .
6: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor PtrDscls3: ., .
7: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SSC1
8: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 2 added in variable: SSC2
9: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 3 added in variable: SSC3
10: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 4 added in variable: SSC4
11: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 5 added in variable: SSC5
12: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 6 added in variable: SSC6
13: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 7 added in variable: SSC7
14: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: PWR1
15: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: STIM1
16: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SLFDIR1
17: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: BENO1
18: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: TRAD1
19: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: CONFM1
20: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SEC1
21: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: PWR2
22: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: ACHV2
23: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: HEDO2
24: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: STIM2
25: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SLFDIR2
26: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: BENO2
27: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: TRAD2
28: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: CONFM2
29: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SEC2
30: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0 added in variable: SMT1
31: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 2 added in variable: SMT2
32: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT2: 2
33: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 3 added in variable: SMT3
34: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT3: 3
35: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 4 added in variable: SMT4
36: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT4: 4
37: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 5 added in variable: SMT5
38: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT5: 5
39: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 6 added in variable: SMT6
40: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT6: 6
41: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Undeclared level(s) 0, 7 added in variable: SMT7
42: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor SMT7: 7
43: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELSENS1: .
44: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELSENS2: .
45: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELSENS3: .
46: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELSENS4: .
47: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELPROC1: .
48: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELPROC2: .
49: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELPROC3: .
50: In read.spss(file = "~/downloads/FRIENDS Combined SME.sav") :
  Duplicated levels in factor AELRESP1: .
```

**Comment by Saguaroman** [comment]
Thanks!  Yes -- exporting to csv works.  

The file does work fine in SPSS, and also, my colleague has been working with the file in R without problems.  I'll investigate further some of those errors above and see what I can find in those specific variables that might be causing problems.  Anyhow, the CSV conversion is an easy workaround; not sure why I didn't think of it!  Thanks for your help.

**Comment by AlexanderLyNL** [comment]
I converted the sav file to csv via R. The errors relate to how R reads the spss file. It might be better to convert the sav file in SPSS to csv directly. Could you try that as well? And see if runs in JASP?

**Comment by Saguaroman** [comment]
Yes -- I did that and it runs fine in JASP.  The errors seem to be caused by value labels in SPSS.  We have some variables that are Likert scales and so have value labels for the poles (1= strongly disagree, 5 = strongly agree) but the intermediate points all just have a period (.) as the value label. Those are being flagged above as "duplicate level."  Then there are some variables where one value is labelled but others are not (e.g., a dichotomous variable where "1" is labelled, but "0" is not because 0 is just the absence of "1"...  apparently if one value has a label, all of them need to. 

Are the error messages above from R or from JASP?  There doesn't seem to be a correspondence between what's causing those issues and the "String too long" message that JASP was giving.  

Either way -- the CSV workaround definitely works.   Thanks!

**Comment by JorisGoosen** [comment]
The above error messages were from R. JASP just seems to go wrong somewhere on appending a string but it was pretty hard to follow the error-message to it's root-cause.

**Comment by JorisGoosen** [comment]
I'll now close this issue as you can now continue using your data with JASP and we are planning a rewrite of loading data into JASP anyway.

**Comment by AlexanderLyNL** [comment]
Thanks for the feedback, as it will be helpful for when we do the rewrite.
